x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -59,6 +59,11 @@ public IndicesPermission indices() {
         throw new UnsupportedOperationException("cannot retrieve indices permission on limited role");
     }
 
+    @Override
+    public RemoteIndicesPermission remoteIndices() {
```

Comment:
We will likely need to support this to support QC-side "regular" API keys being used to make CCS queries but I'd like to move iteratively and change things when we get to serializing index privileges to sent to the FC.

Reply:
Why would we need it for that? I don't quite follow.

Reply:
It depends on how we determine what privileges to send from the QC to the FC. One way is to use the Role instance we have stored as authz info on the QC when we run a CCS action, get `RemoteIndicesPermission` of that role and translate it into a list of `IndicesPrivileges`, and send that on to the FC. When dealing with a QC-side API key, the role might be a `LimitedRole`, which is why we might need to build support here.

Reply:
I don't think we want to do that for this purpose. We already have [a way](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java#L357) to retrieve the role descriptors for a Subject. Using Subject (and Authentication) is likely a better code flow since it is passed around almost everywhere. It also frees us from worrying about custom AuthorizationEngine.

Reply:
Using `getRoleDescriptorsList` is for sure possible, and may be cleaner than using authz info. Will definitely have more discussion around this once we get to that part of the implementation.